src/main/java/net/javadiscord/javabot/systems/help/AutoCodeFormatter.java

```diff
@@ -0,0 +1,164 @@
+	 * setting its methods to be strictly called after {@link HelpListener}.
+	 *
+	 * @param event          a {@link MessageReceivedEvent}
+	 * @param isFirstMessage flag that should be set if the message is a thread-opening one.
+	 */
+	void handleMessageEvent(@Nonnull MessageReceivedEvent event, boolean isFirstMessage) {
+		if (!event.isFromGuild()) {
+			return;
+		}
+		if (event.getAuthor().isBot() || event.getMessage()
+				.getAuthor()
+				.isSystem()) {
+			return;
+		}
+		if (autoMod.hasSuspiciousLink(event.getMessage()) ||
+				autoMod.hasAdvertisingLink(event.getMessage())) {
+			return;
+		}
+		if (event.isWebhookMessage()) {
+			return;
+		}
+		if (!event.isFromThread()) {
+			return;
+		}
+		if (event.getChannel()
+				.asThreadChannel()
+				.getParentChannel()
+				.getIdLong() != botConfig.get(event.getGuild())
+				.getHelpConfig()
+				.getHelpForumChannelId()) {
+			return;
+		}
+		if (!Boolean.parseBoolean(preferenceService.getOrCreate(Objects.requireNonNull(event.getMember())
+				.getIdLong(), Preference.FORMAT_UNFORMATTED_CODE).getState())) {
+			return;
+		}
+
+
+		if (event.getMessage().getContentRaw().contains("```")) {
+			return; // exit if already contains codeblock
+		}
+
+		CodeBlock code = findCodeblock(event);
+		if (code == null) {
+			return;
+		}
+
+		if (isFirstMessage || !event.getMessage().getMentions().getUsers().isEmpty() ||
+				!event.getMessage().getMentions().getRoles().isEmpty() ||
+				event.getMessage().getMentions().mentionsEveryone()) {
+			sendFormatHint(event);
+		} else {
+			replaceUnformattedCode(event.getMessage()
+					.getContentRaw(), code.startIndex(), code.endIndex(), event);
+		}
+	}
+
+	private void sendFormatHint(MessageReceivedEvent event) {
+		event.getMessage()
+				.replyEmbeds(formatHintEmbed(event.getGuild()))
+				.queue();
+	}
+
+	private void replaceUnformattedCode(String msg, int codeStartIndex, int codeEndIndex, MessageReceivedEvent event) {
+		// default case: a "normal", non-ping containing, non first message of a forum-thread containing "{" and "}".
+		// user must also have set their preferences to allow this.
+		if (msg.length() > 1992) { // can't exceed discord's char limit
+			sendFormatHint(event);
+			return;
+		}
+		String messageContent = msg.substring(0, codeStartIndex) + " ```" +
+				msg.substring(codeStartIndex, codeEndIndex) + " ```" + msg.substring(codeEndIndex);
+		EmbedBuilder autoformatInfo = new EmbedBuilder().setDescription(botConfig.get(event.getGuild())
+				.getHelpConfig()
+				.getAutoFormatInfoMessage());
+		WebhookUtil.ensureWebhookExists(
+				event.getChannel()
+						.asThreadChannel()
+						.getParentChannel()
+						.asForumChannel(),
+				wh -> WebhookUtil.replaceMemberMessage(
+						wh,
+						event.getMessage(),
+						messageContent,
+						event.getChannel()
+								.getIdLong(),
+						autoformatInfo.build(),
+						formatHintEmbed(event.getGuild())
```

Comment:
I feel like adding both embeds to the message is too much and that `autoformatInfo` is sufficient. After all, it should be easy to read all the messages and that embed could get in the way.

Reply:
fair